lib/mutant/cli.rb

```diff
@@ -86,7 +86,7 @@ def parse(arguments)
     def parse_match_expressions(expressions)
       fail Error, 'No expressions given' if expressions.empty?
 
-      expressions.map(&Expression.method(:parse)).each do |expression|
+      expressions.map(&config.expression_parser.method(:call)).each do |expression|
```

Comment:
Instead of unnecessarily creating an intermediary array, why not do this:

```ruby
expressions.each do |expression|
  add_matcher(:match_expressions, config.expression_parser.call(expression))
end
```